EIPS/eip-6780.md

```diff
@@ -27,37 +27,50 @@ The behaviour of `SELFDESTRUCT` is changed in the following way:
- - Transfer the account balance to the target **and** set account balance to `0.`
- - Note that no refund is given since [EIP-3529](./eip-3529.md).
- - Note that the rules of [EIP-2929](./eip-2929.md) regarding `SELFDESTRUCT` remain unchanged.
- - Note that when verkle tries are implemented on Ethereum, the cleared storage will be marked as having been written before but empty. This leads to no observable differences in EVM execution, but a contract having been created and deleted will lead to different state roots compared to the action not happening.
+   - `SELFDESTRUCT` continues to behave as previously specified, this includes the following actions
+     - The current execution frame halts.
+     - `SELFDESTRUCT` deletes data as previously specified.
+     - `SELFDESTRUCT` transfers the entire account balance to the target
+     - The account balance of the contact calling `SELFDESTRUCT` is set to `0`.
+   - Note that if the target is the same as the contract calling `SELFDESTRUCT` that ether will be burnt.
+   - Note that no refund is given since [EIP-3529](./eip-3529.md).
+   - Note that the rules of [EIP-2929](./eip-2929.md) regarding `SELFDESTRUCT` remain unchanged.
+   - Note that when verkle tries are implemented on Ethereum, the cleared storage will be marked as having been written before but empty. This leads to no observable differences in EVM execution, but a contract having been created and deleted will lead to different state roots compared to the action not happening.
+
+A contract is considered created at the beginning of a create transaction or when a CREATE series operation begins execution (CREATE, CREATE2, and other operations that deploy contracts in the future).  If a balance exists at the contract's new address it is still considered to be a contract creation. 
 
 The `SELFDESTRUCT` opcode remains deprecated as specified in [EIP-6049](./eip-6049.md). Any use in newly deployed contracts is strongly discouraged even if this new behaviour is taken into account, and future changes to the EVM might further reduce the functionality of the opcode.
 
 ## Rationale
 
 Getting rid of the `SELFDESTRUCT` opcode has been considered in the past, and there are currently no strong reasons to use it. This EIP implements a behavior that will attempt to leave some common uses of `SELFDESTRUCT` working, while reducing the complexity of the change on EVM implementations that would come from contract versioning. A further option that was considered was to just remove storage clearing from `SELFDESTRUCT` in [EIP-6046](./eip-6046.md) while preserving removal of contracts, however this is not safe because existing contracts rely on storage being empty when they are deployed.
 
+Removing "hollow" accounts at addresses that are not externally owned and that only held a balance prior to contract creation provides a better user experience than leaving an account with a non-zero nonce, possible storage, and no code to access that storage.
```

Comment:
I don't understand this sentence. Is it necessary?

Reply:
It covers a specific use case, so it is needed.

Reply:
I was saying it is quite unintelligible. What's a "hollow account"? Do you mean counterfactual accounts?

Can you maybe try rewriting it because it makes no sense to me at all. Can you explain the use case?

Reply:
Counterfactual state channels is one use case. It could just as easily be used with smart contract accounts for stealth addresses.  I'll re-write it using counterfactual accounts as the sample case.